FreeAPSWatch WatchKit Extension/Views/NutrientsView.swift

```diff
@@ -0,0 +1,192 @@
+                    WKInterfaceDevice.current().play(.click)
+                    watchStateModel.addNutrients([
+                        .carb: carbCount,
+                        .protein: proteinCount,
+                        .fat: fatCount
+                    ])
+                } label: {
+                    Text("Add").opacity(isSubmitButtonDisabled ? 0.4 : 1)
+                }
+                .tint(.loopYellow)
+                .disabled(isSubmitButtonDisabled)
+            }
+            .padding([.leading, .trailing], 4)
+            .padding(.bottom, 12)
+        }
+        .frame(maxHeight: .infinity)
+        .edgesIgnoringSafeArea(.bottom)
+        .navigationTitle("Nutrients")
+        .onAppear {
+            if let unwrappedCarbsRequired = watchStateModel.carbsRequired {
+                carbCount = Int(unwrappedCarbsRequired)
+            }
+        }
+    }
+
+    struct GramsPicker: View {
+        @Environment(\.sizeCategory) var sizeCategory
+        @State var step = Constants.defaultPickerStep
+        let labelFrameWidth = WKInterfaceDevice.current().screenBounds.width - 8
+
+        @Binding var selection: Int
+        let label: String
+        let labelAlignment: Alignment
+        let max: Int
+        let accentColor: Color
+
+        var fontForSizeCategory: Font {
+            switch sizeCategory {
+            case .extraSmall,
+                 .small:
+                return .system(size: 12)
+            case .medium:
+                return .system(size: 14)
+            case .large:
+                return .system(size: 16)
+            default:
+                return .system(size: 20)
+            }
+        }
+
+        var fontForSizeCategoryForHundreds: Font {
+            switch sizeCategory {
+            case .extraSmall:
+                return .system(size: 8)
+            case .small:
+                return .system(size: 9)
+            case .large,
+                 .medium:
+                return .system(size: 14)
+            default:
+                return .system(size: 16)
+            }
+        }
+
+        var body: some View {
+            GeometryReader { geometry in
+                Picker(selection: $selection) {
+                    ForEach(0 ... (max / step), id: \.self) {
+                        let value = $0 * step
+                        Text("\(value) g")
+                            .tag(value)
+                            .font(value < 99 ? fontForSizeCategory : fontForSizeCategoryForHundreds)
+                            .foregroundColor(accentColor)
+                    }
+                } label: {
+                    var labelFrameTranslationX: CGFloat {
+                        let offset = geometry.size.width + Constants.pickerHSpacing
+                        if labelAlignment == .leading { return offset }
+                        if labelAlignment == .trailing { return -offset }
+                        return 0
+                    }
```

Comment:
In order to give more space to each label. By default the label is limited by the width of its `Picker`.

<img width="546" alt="Screenshot 2023-07-02 at 21 44 25" src="https://github.com/Artificial-Pancreas/iAPS/assets/17122466/829557d6-62c6-44cd-b36e-e2290177a0c6">